src/bridges/nft-basic/NftVault.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Aztec.
+pragma solidity >=0.8.4;
+
+import {IERC721} from "../../../lib/openzeppelin-contracts/contracts/interfaces/IERC721.sol";
+import {AztecTypes} from "../../../lib/rollup-encoder/src/libraries/AztecTypes.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {AddressRegistry} from "../registry/AddressRegistry.sol";
+
+/**
+ * @title Basic NFT Vault for Aztec.
+ * @author Josh Crites, (@critesjosh on Github),  Aztec Team
+ * @notice You can use this contract to hold your NFTs on Aztec. Whoever holds the corresponding virutal asset note can withdraw the NFT.
+ * @dev This bridge demonstrates basic functionality for an NFT bridge. This may be extended to support more features.
+ */
+contract NftVault is BridgeBase {
+    struct NftAsset {
```

Comment:
Do you need full `uint256`? If you can handle all expect token ids with a `uint96` the `NftAsset` structure can be packed into one storage slot saving you a `sstore`.

Reply:
tokenId is typically of type `uint256` in erc721 contracts and since this contract is generic and should work with all erc721 contract, and we don't know what the possible range of tokenIds will be for a given erc721 contract I figured I should use `uint256`